DOC Update security issues docs (#371)
silverstripe/developer-docs: en/05_Contributing/17_Managing_Security_Issues.md

Silverstripe staff are granted access to review undisclosed security issues on a need-to-know basis and are bound by their employment contracts. Core committers not currently employed by Silverstripe are allowed to review undisclosed security issues after signing a non-disclosure agreement.

## Handling security issues confidentially

This process is relevant when a potential vulnerability is reported confidentially and the Silverstripe CMS development team is in a position to prepare a patch prior to the public disclosure of the vulnerability.

This process is usually started once someone [reports a security issue](issues_and_bugs/#reporting-security-issues).

### When receiving a report

[hint]
Ensure the reporter is given a justification for all issues we conclude are not security vulnerabilities.
[/hint]

1. An automated response is sent back to the reporter to acknowledge receipt of their vulnerability report.

> 1. Perform an initial assessment of the report. If it is obviously spam or otherwise invalid, reply to the reporter thanking them for their efforts and explaining that we won't be handling this as a security issue.

Reviewer: ```suggestion
1. Perform an initial assessment of the report. If the report does not qualify as a security issue then reply to the reporter thanking them for their efforts and explaining that we won't be handling this as a security issue.
```

Reviewer 2: Done, but I kept the comma instead of "then" to be more natural.

    - If what was raised is a valid bug but not a valid security report, advise them to raise an issue on GitHub.
    - If the report was spam, see [spam](#spam) below.
1. If the report is valid, [create a issue in our private security repository](https://github.com/silverstripe-security/security-issues/issues/new).
    - Prefix the issue name with "[NO CVE REQUESTED YET]".
    - If encrypted information or attachments are provided, attach them to the private security issue.
    - Add the new issue to the "Triage" column on the [project board](https://github.com/silverstripe-security/security-issues/projects/1).
1. Reply to [security@silverstripe.org](mailto:security@silverstripe.org) ONLY (i.e. do not include the reporter in this reply) and provide a link to the private security issue. Keep most of the discussion on GitHub.

> 1. Clarify who picks up and owns the issue (assign in GitHub).

Reviewer: ```suggestion
1. Clarify who picks up the issue for an initial assessment (assign in GitHub).
```

Reviewer 2: The idea is more than just doing the initial assessment - they're the person who communicates with the reporter, applies for a CVE, etc.

Reviewer: ok, "picks up" and "owns" are the same thing, so get rid of " and owns"

Reviewer 2: Done and added a clarification similar to my comment above to avoid ambiguity.

>     - The owner can be separate from the developer resolving the issue. Their primary responsibility is to ensure the issue keeps moving through the process correctly.

Reviewer: ```suggestion
    - The developer doing an initial assessment on the issue can be separate from the developer implementing a fix for the issue.
```

Reviewer 2: Reworded a bit, since this suggestion assumed the person who picks up the issue was only doing the initial assessment.

> 1. Perform a criticality assessment to determine how severe the issue is and if it can be replicated.

Reviewer: ```suggestion
1. Perform an assessment to determine how severe the issue is and if it can be replicated.
```

Reviewer 2: Yeah I didn't like "criticality" when it was first added either lol.
Done.

  - **Ecosystem** should always be `composer`.
  - **Package name** must match the package name in Packagist.
  - **Affected versions** must be a string representing a range of versions affected by the vulnerability.
  - **Patched versions** must list the versions where the vulnerability was/will be patched.
- **Severity** must be set to _Assess severity using CVSS_ and the CVSS option must be specified in the calculator.
- **Common weakness enumerator (CWE)** should be left blank if no suitable option can be found in the list.
- **CVE identifier** must be set to "Request CVE ID later" until we have had a CVE assigned.
- **Title** and **Description** should be descriptive enough that people can evaluate the risk without being so specific that would-be-hackers can start exploiting them right away.

These settings can be changed later, so if you are missing information when you first create the draft you can come back to add the information later.

### Developing a fix

- Move the issue into "In Progress" on the [project board](https://github.com/silverstripe-security/security-issues/projects/1)

> - If the silverstripe-security GitHub organisation doesn't have a pre-existing private repo for the module that needs to be fixed, create one and push the public git history to it. You cannot create a private fork of a public repo.

Reviewer: ```suggestion
- If the silverstripe-security GitHub account doesn't have a pre-existing private repo for the module that needs to be fixed, do not create a fork because you cannot create a private fork of a private repo. Instead create a new repo on the silverstripe-security account and push the public git history to it.
```

Reviewer 2: Agreed with all except changing "organisation" to "account" - it's not an account like mine or yours. It's an organisation in GitHub terminology.

Reviewer 2: Done, but also left public as public - the point is that we can't make a private security fork of our public core/supported modules.
